### PR TITLE
gh-113008:  Correct argparse usage output for required, mutually exclusive groups

### PR DESCRIPTION
**Pull request by paytonward6**
Prior to fix,  a mutually exclusive group that contained a positional would not be printed properly when the positional is not defined first. Instead, it would completely ignore the mutual exclusiveness in the usage output.

For example:
```python3
parser = ArgumentParser(prog="PROG")
group = parser.add_mutually_exclusive_group(required=True)
group.add_argument("--foo")
group.add_argument("bar", nargs="?")
parser.parse_args()
```
would have a usage line of:
> usage: PROG [-h] (--foo FOO | bar)

while
```python3
parser = ArgumentParser(prog="PROG")
group = parser.add_mutually_exclusive_group(required=True)
group.add_argument("bar", nargs="?")
group.add_argument("--foo")
parser.parse_args()
```
would have a usage line of:
> usage: PROG [-h] [--foo FOO] [bar]

* Issue: gh-113008

### CODE REVIEW

File: Lib/test/test_argparse.py [hunk 1]
```diff
@@ -2836,6 +2836,29 @@ def test_help(self):
               '''
         self.assertEqual(parser.format_help(), textwrap.dedent(expected))
 
+    def test_optional_order(self):
+        parser = ErrorRaisingArgumentParser(prog="PROG")
+        group = parser.add_mutually_exclusive_group(required=True)
+        group.add_argument("-", dest="FOO")
```
Comment (serhiy-storchaka) [inline Lib/test/test_argparse.py]: "-" is a very uncommon case. I think that you can test this with '--foo'.

BTW, use single quotes, they are more common in this file.

File: Lib/argparse.py [hunk 2]
```diff
@@ -398,13 +398,17 @@ def _format_actions_usage(self, actions, groups):
                 raise ValueError(f'empty group {group}')
 
             try:
-                start = actions.index(group._group_actions[0])
+                start = _sys.maxsize
+                for item in group._group_actions:
+                    index = actions.index(item)
+                    if index < start:
+                        start = index
```
Comment (serhiy-storchaka) [inline Lib/argparse.py]: I think that you can use `min()`:
```suggestion
                start = min(actions.index(item) for item in group._group_actions)
```
or
```suggestion
                start = min(map(actions.index, group._group_actions))
```

**Review by serhiy-storchaka — APPROVED** [review]
Thank you @paytonward6. LGTM.

**Comment by serhiy-storchaka** [comment]
There are some conflicts. Please merge the `main` branch into your branch.

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution @paytonward6.

**Comment by paytonward6** [comment]
Thanks for the help @serhiy-storchaka, I appreciate it

**Pull request merged**